src/sparse_blas/common_op_verification.hpp

```diff
@@ -0,0 +1,142 @@
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+#ifndef _ONEMKL_SRC_SPARSE_BLAS_COMMON_OP_VERIFICATION_HPP_
+#define _ONEMKL_SRC_SPARSE_BLAS_COMMON_OP_VERIFICATION_HPP_
+
+#include <string>
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+#include "oneapi/mkl/sparse_blas/types.hpp"
+#include "macros.hpp"
+
+namespace oneapi::mkl::sparse::detail {
+
+/// Throw an exception if the scalar is not accessible in the host
+inline void check_ptr_is_host_accessible(const std::string &function_name,
+                                         const std::string &scalar_name,
+                                         bool is_ptr_accessible_on_host) {
+    if (!is_ptr_accessible_on_host) {
+        throw mkl::invalid_argument(
+            "sparse_blas", function_name,
+            "Scalar " + scalar_name + " must be accessible on the host for buffer functions.");
+    }
+}
+
+template <typename InternalSparseMatHandleT>
+void check_valid_spmm_common(const std::string &function_name,
+                             oneapi::mkl::sparse::matrix_view A_view,
+                             InternalSparseMatHandleT internal_A_handle,
+                             oneapi::mkl::sparse::dense_matrix_handle_t B_handle,
+                             oneapi::mkl::sparse::dense_matrix_handle_t C_handle,
+                             bool is_alpha_host_accessible, bool is_beta_host_accessible) {
+    THROW_IF_NULLPTR(function_name, internal_A_handle);
+    THROW_IF_NULLPTR(function_name, B_handle);
+    THROW_IF_NULLPTR(function_name, C_handle);
+
+    check_all_containers_compatible(function_name, internal_A_handle, B_handle, C_handle);
+    if (internal_A_handle->all_use_buffer()) {
+        check_ptr_is_host_accessible("spmm", "alpha", is_alpha_host_accessible);
+        check_ptr_is_host_accessible("spmm", "beta", is_beta_host_accessible);
+    }
+    if (is_alpha_host_accessible != is_beta_host_accessible) {
+        throw mkl::invalid_argument(
+            "sparse_blas", function_name,
+            "Alpha and beta must both be placed on host memory or device memory.");
+    }
+    if (B_handle->dense_layout != C_handle->dense_layout) {
+        throw mkl::invalid_argument("sparse_blas", function_name,
+                                    "B and C matrices must used the same layout.");
+    }
+
+    if (A_view.type_view != oneapi::mkl::sparse::matrix_descr::general) {
+        throw mkl::invalid_argument("sparse_blas", function_name,
+                                    "Matrix view's type must be `matrix_descr::general`.");
```

Comment:
```suggestion
                                    "Matrix view's `type_view` must be `matrix_descr::general`.");
```

Reply:
Reworded in https://github.com/oneapi-src/oneMKL/pull/527/commits/7aa5177635e730a6f394783d996912de40a7be7c